src/lib/operators/table_scan/sorted_segment_between_search.hpp

```diff
@@ -0,0 +1,203 @@
+#pragma once
+
+#include <functional>
+#include <thread>
+#include <type_traits>
+
+#include <boost/range.hpp>
+#include <boost/range/join.hpp>
+
+#include "all_type_variant.hpp"
+#include "constant_mappings.hpp"
+#include "types.hpp"
+
+namespace opossum {
+
+// Generic class which handles the actual scanning of a sorted segment
+template <typename IteratorType, typename SearchValueType>
+class SortedSegmentBetweenSearch {
+ public:
+  SortedSegmentBetweenSearch(IteratorType begin, IteratorType end, const OrderByMode& order_by, const bool nullable,
```

Comment:
Let's take a look on this file tomorrow. At first sight, it looks like a copy with few modifications of `sorted_segment_search.hpp`.

Reply:
Yes, that was indeed our initial approach.

Reply:
Makes sense for development to start off quick, but we should make sure that we don't create unnecessary copies.
There is also another reason, we can talk about tomorrow.